fhirpipe/transform/fhir.py

```diff
@@ -0,0 +1,120 @@
+                col_name = new_col_name(inp["script"], col_name)
+            cols_to_fetch[0].append(col_name)
+
+        # Else retrieve the static value
+        else:
+            cols_to_fetch[1].append(inp["staticValue"])
+
+    if attribute_structure["mergingScript"]:
+        result = row[new_col_name(attribute_structure["mergingScript"], cols_to_fetch)]
+    else:
+        result = [row[c] for c in cols_to_fetch[0]]
+        result.extend(cols_to_fetch[1])
+
+    # Unlist
+    if len(result) == 1:
+        result = result[0]
+
+    return result
+
+
+def rec_create_fhir_object(fhir_obj, attribute_structure, row):
+    if isinstance(attribute_structure, dict):
+        # If there are some inputs, we can build the objects to output
+        if "inputs" in attribute_structure.keys() and attribute_structure["inputs"]:
+            result = build_fhir_attribute(attribute_structure, row)
+            fhir_obj[attribute_structure["name"]] = result
+
+        # Otherwise, we can recurse on children
+        else:
+            is_array = attribute_structure["fhirType"] == "array"
+
+            fhir_obj[attribute_structure["name"]] = [] if is_array else dict()
+
+            for attr in attribute_structure["children"]:
+                if is_array:
+                    child = {}
+                    rec_create_fhir_object(child, attr, row)
+                    child = list(child.values())
+                    fhir_obj[attribute_structure["name"]].append(child)
+                else:
+                    rec_create_fhir_object(
+                        fhir_obj[attribute_structure["name"]], attr, row
+                    )
+
+            # TODO bind references only after having loading all the resource?
+            # If the object is a Reference, to we give it to bind_reference
+            if attribute_structure["fhirType"].startswith("Reference"):
+                bind_reference(fhir_obj[attribute_structure["name"]])
+
+    # If the current object is a list, we can repeat the same steps as above for each item
+    elif isinstance(fhir_obj, list) and len(fhir_obj) > 0:
+        children = []
+        for child_spec in fhir_obj:
+            children.append(rec_create_fhir_object(dict(), child_spec, row))
+
+        fhir_obj[attribute_structure["name"]] = children
+
+
+def bind_reference(fhir_object):
+    """
+    Analyse a reference and replace the provided identifier
+    with official FHIR uri if the resource reference exists.
+
+    args:
+        fhir_object: the fhir object to parse
+
+    return:
+        a fhir_object where valid references are now FHIR uris
+    """
+    # TODO think again about how to bind references
+
+    # First we check that the reference has been provided
+    if fhir_object and fhir_object["identifier"]["value"]:
+        # get id
+        identifier = fhir_object["identifier"]["value"]
+        # get referenced type (URI)
+        uri = fhir_object["identifier"]["system"]
```

Comment:
shouldn't the uri be the fhirType right now ?

Reply:
I don't think so, when we reach this function, fhir_object has the fhirType reference and we want to know where to look for the reference. This location is provided in identifier.system (as for the last discussion about references but this could be incorrect)

Reply:
I'm having troubles understanding how this works with the find_fhir_resource() function below. We'll discuss this in person :)